File: javawumpus/test/WumpusTest.java
import static org.junit.Assert.*;

import java.util.Random;

import org.junit.Before;
import org.junit.Test;


public class WumpusTest {

	private Wumpus testObj;

	@Before
	public void setUp() {
		testObj = new Wumpus();
		testObj.selector = new Selector(new Random(0)); //seed random for expected results
	}
		
	class WumpusWithSetMapItems extends Wumpus {
		public WumpusWithSetMapItems(int[] mapItems) {
			this.items = new MapItemLocations(mapItems);
		}
	}
	
	@Test
	public void testCrossover() throws Exception {
		// remember zeroth index is not used
		int[] withDups = {0,1,1,2,3,4,5};
		Wumpus testObj = new WumpusWithSetMapItems(withDups);
		assertTrue(testObj.items.crossover());
		int[] withDupsAtEnds = {0,1,2,3,4,5,1};
		testObj = new WumpusWithSetMapItems(withDupsAtEnds);
		assertTrue(testObj.items.crossover());
		int[] unique = {0,1,2,3,4,5,6};
		testObj = new WumpusWithSetMapItems(unique);
		assertFalse(testObj.items.crossover());
	}

}

File: CSharpWumpus/Wumpus/WumpusTest.cs
using System;
using NUnit.Framework;

namespace Wumpus
{
    [TestFixture]
    public class ProgramTest
    {
        [SetUp]
        public void SetUp()
        {
            Game.random = new Random(0); //seed random for expected results
        }

        [TestCase]
        public void fnA()
        {
            Assert.AreEqual(15, Game.fnA());
            Assert.AreEqual(17, Game.fnA());
            Assert.AreEqual(16, Game.fnA());
            Assert.AreEqual(12, Game.fnA());
            Assert.AreEqual(5, Game.fnA());
            Assert.AreEqual(12, Game.fnA());
            Assert.AreEqual(19, Game.fnA());
            Assert.AreEqual(9, Game.fnA());
            Assert.AreEqual(20, Game.fnA());
            Assert.AreEqual(6, Game.fnA());
        }

        [TestCase]
        public void fnB()
        {
            Assert.AreEqual(3, Game.fnB());
            Assert.AreEqual(3, Game.fnB());
            Assert.AreEqual(3, Game.fnB());
            Assert.AreEqual(2, Game.fnB());
            Assert.AreEqual(1, Game.fnB());
            Assert.AreEqual(2, Game.fnB());
            Assert.AreEqual(3, Game.fnB());
            Assert.AreEqual(2, Game.fnB());
            Assert.AreEqual(3, Game.fnB());
            Assert.AreEqual(1, Game.fnB());
        }

        [TestCase]
        public void fnC()
        {
            Assert.AreEqual(3, Game.fnC());
            Assert.AreEqual(4, Game.fnC());
            Assert.AreEqual(4, Game.fnC());
            Assert.AreEqual(3, Game.fnC());
            Assert.AreEqual(1, Game.fnC());
            Assert.AreEqual(3, Game.fnC());
            Assert.AreEqual(4, Game.fnC());
            Assert.AreEqual(2, Game.fnC());
            Assert.AreEqual(4, Game.fnC());
            Assert.AreEqual(2, Game.fnC());
        }
    }
}
